UI/SidePanel.cpp

```diff
@@ -3211,7 +3211,9 @@ void SidePanel::DoLayout() {
 
     // system name / droplist
     ul = GG::Pt(GG::X(MaxPlanetDiameter()), GG::Y0);
-    lr = ul + GG::Pt(ClientWidth() - GG::X(MaxPlanetDiameter()), name_height);
+    auto system_name_width = ClientWidth() - GG::X(MaxPlanetDiameter());
```

Comment:
Thanks for taking the time to review this PR.

Yes, typing X would have been less text that `auto`.

However, the type inference from `auto` provides more benefits than saved typing:
- `auto` variables are always initialized.  The type inference of `auto` only occurs if it has a constructor from which to infer the type.  `SomeType x;` may/may not be initialized, but `auto x = <anything that compiles>` is always initialized.  Using `auto` protects against unintentionally uninitialized variables.
- `auto` variables are always just constructed.  They are not constructed and then moved or copied. 
- pointer/reference `auto` variables are never aliased/sliced down to a base type, even if the type on the right hand side changes in the future after the code is written.  The type is correctly inferred at compile time.
- `auto` variables will not call implicit type converting constructors or casts.

For these reasons `auto` is more letters than `X`, but better code.